### ISSUE TITLE
Support SVG logos

### ISSUE THREAD

**Issue by astorije** [opencollective/opencollective#964]
It would be really nice if you could support SVG files for logo and cover image in https://opencollective.com/[...]/edit#images.

I tried uploading an SVG logo, and the spinner never stopped, so I had to reload the page and try my luck with a PNG :)

Thanks! 💅

**Comment by xdamman** [comment]
Yes it would be nice. But that's a bit more involved. I'll add the "help wanted" tag if someone wants to take it on.

**Comment by Betree** [comment]
_Recently requested in https://opencollective.freshdesk.com/a/tickets/9760_

To add a bit more context on this topic, we can't just make a one line change to allow SVG uploads, there are a few security concerns that we need to address to allow that such as [XEE processing attacks](https://owasp.org/www-community/vulnerabilities/XML_External_Entity_(XXE)_Processing). [The W3C](https://www.w3.org/TR/SVGTiny12/mimereg.html) lists the main attack vectors on SVGs. A similar [ticket](https://core.trac.wordpress.org/ticket/24251) for wordpress has been pending for 7 years now because of these security issues (it's a good resource to learn more about this topic).

We may be able to use libraries such as https://github.com/cure53/DOMPurify to prevent these attacks.

**Comment by scenaristeur** [comment]
Hi 👋
Exploring OC code, and the big list of issue I'm wondering if it should be possible to convert the SVG to PNG when uploading so there is no more XML to allow attack, something like https://www.npmjs.com/package/convert-svg-to-png . But that is just an idea, I'm not really sure of the context.

**Comment by Betree** [comment]
> Hi wave Exploring OC code, and the big list of issue I'm wondering if it should be possible to convert the SVG to PNG when uploading so there is no more XML to allow attack, something like https://www.npmjs.com/package/convert-svg-to-png . But that is just an idea, I'm not really sure of the context.

That's a good option. Our [images service](https://github.com/opencollective/opencollective-images) already converts images on the fly so we could re-use that feature.

Note that this issue is quite complex, we don't recommend working on something like this for a first contribution. You can try this [list](https://github.com/opencollective/opencollective/issues?q=is%3Aopen+is%3Aissue+no%3Aassignee+label%3A%22complexity+%E2%86%92+minimal%22%2C%22complexity+%E2%86%92+simple%22+-label%3Ateam%2Cdiscussion%2C%22blocked+%E2%86%92+waiting+third+party%22%2C%22blocked+%E2%86%92+waiting+parent%22%2C%22blocked+%E2%86%92+waiting+feedback%22%2C%22needs+design%22%2C%22needs+specs%22+) if you're looking for that, or you can ask on https://slack.opencollective.com/ > `#engineering`.

**Comment by Dreamsorcerer** [comment]
Aren't we just uploading a file and then serving it as an img tag in a webpage? Surely the browser will handle these security concerns (possibly with a CSP to tighten it up further and disallow access to remote resources)?
